src/Core/Settings.cpp

```diff
@@ -4453,7 +4453,7 @@ Optimize GROUP BY when all keys in block are constant
     M(Bool, legacy_column_name_of_tuple_literal, false, R"(
 List all names of element of large tuple literals in their column names instead of hash. This settings exists only for compatibility reasons. It makes sense to set to 'true', while doing rolling update of cluster from version lower than 21.7 to higher.
 )", 0) \
-    M(Bool, enable_named_columns_in_function_tuple, false, R"(
+    M(Bool, enable_named_columns_in_function_tuple, true, R"(
 Generate named tuples in function tuple() when all names are unique and can be treated as unquoted identifiers.
 Beware that this setting might currently result in broken queries. It's not recommended to use in production
```

Comment:
We enabled it while there is still broken behaviour, or at least I understand so, (https://github.com/ClickHouse/ClickHouse/pull/70853#issuecomment-2433542210) and the description still says it's broken and not to use it in production.